src/ssl.ml

```diff
@@ -301,26 +306,137 @@ end
 
 (* Same as above, but doesn't release the lock. *)
 module Runtime_lock_base = struct
-  external connect : socket -> unit = "ocaml_ssl_connect"
-
-  external accept : socket -> unit = "ocaml_ssl_accept_blocking"
-
-  external write : socket -> Bytes.t -> int -> int -> int
-    = "ocaml_ssl_write_blocking"
-  external write_substring : socket -> string -> int -> int -> int
-    = "ocaml_ssl_write_blocking"
-  external write_bigarray : socket -> bigarray -> int -> int -> int
-    = "ocaml_ssl_write_bigarray_blocking"
-
-  external read : socket -> Bytes.t -> int -> int -> int
-    = "ocaml_ssl_read_blocking"
-
-  external read_into_bigarray :
-    socket -> bigarray -> int -> int -> int = "ocaml_ssl_read_into_bigarray_blocking"
-
-  external flush : socket -> unit = "ocaml_ssl_flush_blocking"
-
-  external ssl_shutdown : socket -> bool = "ocaml_ssl_shutdown_blocking"
+  external raw_connect : socket -> int = "ocaml_ssl_connect_blocking"
+
+  let connect socket =
+    let ret = raw_connect socket in
+    let err = get_error socket ret in
+    if err != Error_none then
```

Comment:
could we pattern match on this variant? Perhaps using a helper function?

Reply:
I went in another direction: no need to call get_error is ret = 1. And if ret <>  1, no need to check if err != Error_none. I updated all functions from Runtime_lock.

In [616434f](https://github.com/savonet/ocaml-ssl/pull/117/commits/616434f1b575558b4bd5c4ab9b099ebc2ea0d00f)

Reply:
I don't mind your idea, but it looks like the logic is backwards. `SSL_ERROR_NONE` is `0`, I'm pretty sure. so you want to raise `if ret <> 0` not `if ret <> 1`, right?

Reply:
I meant test return value as suggested in the documentation before calling get_error. The test depends upon the function but it always clear in the doc.